perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_elixirBonaFideStatus.java

```diff
@@ -0,0 +1,120 @@
+import cz.metacentrum.perun.core.implApi.modules.attributes.UserVirtualAttributesModuleImplApi;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Vojtech Sassmann &lt;vojtech.sassmann@gmail.com&gt;
+ */
+public class urn_perun_user_attribute_def_virt_elixirBonaFideStatus extends UserVirtualAttributesModuleAbstract implements UserVirtualAttributesModuleImplApi {
+
+	private final static Logger log = LoggerFactory.getLogger(urn_perun_user_attribute_def_virt_elixirBonaFideStatus.class);
+
+	private final Pattern addUserExtSource = Pattern.compile("UserExtSource:\\[(.*)] added to User:\\[(.*)]");
+	private final Pattern removeUserExtSource = Pattern.compile("UserExtSource:\\[(.*)] removed from User:\\[(.*)]");
+	private final Pattern extSourceRems = Pattern.compile("cz.metacentrum.perun.core.impl.ExtSourceREMS");
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException {
+		Attribute attribute = new Attribute(attributeDefinition);
+		String url = "http://www.ga4gh.org/beacon/bonafide/ver1.0";
+
+		try {
+			Attribute attributeDef = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF + ":elixirBonaFideStatusREMS");
+			String attrDefValue = (String) attributeDef.getValue();
+			if (attrDefValue != null && !attrDefValue.isEmpty()) {
+				attribute.setValue(attrDefValue);
+			} else {
+				Attribute uesAttribute = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_VIRT + ":eduPersonScopedAffiliations");
+				List<String> affiliations = (List<String>)uesAttribute.getValue();
+				for (String affiliation : affiliations) {
+					if (affiliation != null && affiliation.startsWith("faculty@")) {
+						attribute.setValue(url);
+						break;
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read elixirBonaFideStatusREMS from user " + user.getId(), e);
+		}
+		return attribute;
+	}
+
+	@Override
+	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, String message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<String> resolvingMessages = new ArrayList<>();
+		if (message == null) return resolvingMessages;
+
+		Matcher addUserExtSourceMatcher = addUserExtSource.matcher(message);
+		Matcher removeUserExtSourceMatcher = removeUserExtSource.matcher(message);
+		Matcher extSourceRemsMatcher = extSourceRems.matcher(message);
+
+		User user;
+		Attribute attrVirtRemsLogins;
+
+		if(extSourceRemsMatcher.find()) {
+			if (addUserExtSourceMatcher.find() || removeUserExtSourceMatcher.find()) {
```

Comment:
I am afraid this is not enough. The value of this attribute can be changed by updating of affiliations attribute value (for example if record with "faculty@" was removed).